src/idv_service/session/retrieve/configuration/capture/source/allowed.source.response.js

```diff
@@ -1,4 +1,11 @@
+const Validation = require('../../../../../../yoti_common/validation');
+
 class AllowedSourceResponse {
+  constructor(type) {
+    Validation.isString(type, 'type');
+    this.type = type;
+  }
```

Comment:
So, that is fine, but this is not quite an Abstract class... which is not existing in vanilla js, but can be simulated... aka, the constructor of the Abstract throws an error if the class is instantiated, and therefore only allow its subclasses to be instantiated....
I am not sure of the value of having such Abstract behaviour, but this has been implemented in a few places in this code base.